libs/math/include/math/tensor_slice_iterator.hpp

```diff
@@ -387,16 +387,10 @@ class TensorSliceIterator
         volume = stride[i];
 
         s.volume      = volume;
-        SizeType diff = (s.to - s.from);
-        s.total_steps = diff / s.step;
-        if (s.total_steps * s.step < diff)
-        {
-          ++s.total_steps;
-        }
+        s.total_steps = ((s.to - s.from - 1) / s.step) + 1;
```

Comment:
if you remove the +1 from the Tensor::Slice implementation you can remove the -1 from this equation i think?